service/src/main/java/dk/dbc/opensearch/reponse/SearchProcessorBean.java

```diff
@@ -108,21 +126,62 @@ private void process(SearchResponse response) throws XMLStreamException, IOExcep
     private void compute() {
         String trackingId = request.getTrackingId();
         RepositorySettings repoSettings = getRepoSettings();
-        int start = request.getStartOrDefault();
-        int step = request.getStepValueOrDefault();
+        RepositoryAbstraction abstraction = repoSettings.abstraction();
         ResultSetKey key = ResultSetKey.of(request);
-        resultSet = resultSetCache.get(key);
-        if (resultSet == null)
-            resultSet = makeResultSet(key, repoSettings, trackingId);
+        getResultSet(key, repoSettings, trackingId, abstraction);
 
-        // Make it visible to others while we're working on it
-        // So that others don't make the same object
-        resultSetCache.put(key, resultSet);
-
-        resultSet.fetchWorks(repoSettings.solrClient(),
+        int start = request.getStartOrDefault();
+        int step = request.getStepValueOrDefault();
+        resultSet.fetchWorks(abstraction.getSolrClient(),
                              timings, start, step,
                              trackingId);
+        log.trace("resultSet = {}", resultSet);
+
         resultSetCache.put(key, resultSet);
+
+        fetchRecordsAndFormatThem(step, start, repoSettings, abstraction, trackingId);
+    }
+
+    private void fetchRecordsAndFormatThem(int step, int start, RepositorySettings repoSettings, RepositoryAbstraction abstraction, String trackingId) {
+        for (int i = 0 ; i < step ; i++) {
```

Comment:
Hvorfor ikke dette, og så slette et par linjer i loopet?
```suggestion
        for (int index = start ; index < Integer.min(start + step,resultSet.getHitCount()) ; index++) {
```

Reply:
Forbandede IDE's refactor.... ;-)

Reply:
Jeg vil skulle bruge betydeligt længere på al læse den kode når jeg kommer tilbage til det om 2 mdr, fordi der er noget som gør underligt. I mine øjne - hellere et par linjer ekstra så man kan læse intentionen.

Reply:
Jeg har det lidt omvendt, jo flere variable og breaks man introducerer jo større er state space, og dermed lidt mere uoverskueligt, og muligheder for ting der gør mærkeligt.
Men det er fint, vil ikke påkræve ændringer :smile:

Reply:
Mødes på halvvejen?

int maxIndexInSet = Integer.min(start + step,resultSet.getHitCount())
for (int index = start ; index < maxIndexInSet ; index++) {

Reply:
Nope... resultSet.getHitCount() er ikke statisk.